### PR TITLE
use toFixedAsString and DoubleProperty in diagnosticProperties

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

use `DoubleProperty` instead of `DiagnosticProperty<double>` and use toFixedAsString(1) in place of String interpolation in several cases that were flagged during rolling flutter web.

## Related Issues

https://github.com/flutter/flutter/pull/33349

### CODE REVIEW

File: packages/flutter/lib/src/foundation/diagnostics.dart [hunk 1]
```diff
@@ -2506,6 +2508,15 @@ class DiagnosticsProperty<T> extends DiagnosticsNode {
   /// of the property should be displayed without line breaks if possible.
   String valueToString({ TextTreeConfiguration parentConfiguration }) {
     final T v = value;
+    // Normalize web/DDC specific closure toString syntax.
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/diagnostics.dart]: This change is unrelated to the PR description, no?

Comment (goderbauer) [reply packages/flutter/lib/src/foundation/diagnostics.dart]: Not sure I understand the motivation behind this?

Comment (jonahwilliams) [reply packages/flutter/lib/src/foundation/diagnostics.dart]: Oh woops, yes this one is slightly unrelated.

**Comment by jonahwilliams** [comment]
Updated to use debugFormatDouble

File: packages/flutter/lib/src/foundation/debug.dart [hunk 2]
```diff
@@ -73,3 +74,21 @@ Future<T> debugInstrumentAction<T>(String description, Future<T> action()) {
 const Map<String, String> timelineWhitelistArguments = <String, String>{
   'mode': 'basic',
 };
+
+/// Configure `debugFormatDouble` using [num.toStringAsPrecision].
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/debug.dart]: [debugFormatDouble] here and below.

File: packages/flutter/lib/src/foundation/debug.dart [hunk 3]
```diff
@@ -73,3 +74,21 @@ Future<T> debugInstrumentAction<T>(String description, Future<T> action()) {
...
+///
+/// Defaults to `null`, which uses the default logic of `debugFormatDouble`.
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/debug.dart]: I believe our style is to not enclose null in `

File: packages/flutter/lib/src/painting/matrix_utils.dart [hunk 4]
```diff
@@ -265,9 +265,13 @@ class MatrixUtils {
 List<String> debugDescribeTransform(Matrix4 transform) {
   if (transform == null)
     return const <String>['null'];
-  final List<String> matrix = transform.toString().split('\n').toList();
-  matrix.removeLast();
-  return matrix;
+  final Float64List storage = transform.storage;
+  return <String>[
+    '[0] ${debugFormatDouble(storage[transform.index(0, 0)])},${debugFormatDouble(storage[transform.index(0, 1)])},${debugFormatDouble(storage[transform.index(0, 2)])},${debugFormatDouble(storage[transform.index(0, 3)])}',
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/matrix_utils.dart]: Can you just use `transform.entry(row, column)` instead of getting the index and then accessing storage? Might be a little simpler..

File: packages/flutter/test/foundation/double_precision_test.dart [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    expect(debugFormatDouble(1.1), '1.1');
+    expect(debugFormatDouble(null), 'null');
+  });
+
+  test('debugDoublePrecision can control double precision', () {
+    debugDoublePrecision = 3;
+    expect(debugFormatDouble(1), '1.00');
+    debugDoublePrecision = null;
+  });
+}
```
Comment (goderbauer) [inline packages/flutter/test/foundation/double_precision_test.dart]: nit: add line break?

File: packages/flutter/lib/src/foundation/debug.dart [hunk 6]
```diff
@@ -73,3 +74,21 @@ Future<T> debugInstrumentAction<T>(String description, Future<T> action()) {
...
+int debugDoublePrecision;
+
+/// Formats a double to have standard formatting.
+///
+/// This behavior can be overriden by [debugDoublePrecision].
+String debugFormatDouble(double value) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/debug.dart]: `debug.dart` doesn't really feel like the right place for this since it is not a debug method and called from regular code?

Comment (jonahwilliams) [reply packages/flutter/lib/src/foundation/debug.dart]: Hopefully toDiagnosticsNode and toString are debug code...

**Review by goderbauer — APPROVED** [review]
LGTM
